src/diffusers/models/modeling_utils.py

```diff
@@ -699,6 +699,7 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                             offload_folder=offload_folder,
                             offload_state_dict=offload_state_dict,
                             dtype=torch_dtype,
+                            force_hooks=True,
```

Comment:
why do we nee this?

Reply:
In order to ensure the data placement is handled automatically by `accelerate`. This will be needed because the text encoder might be on a different device and hence its output. So when it hits the UNet (which might be on a different device), the device placements take place seamlessly.